generator/src/test/java/com/scottlogic/deg/generator/restrictions/SetRestrictionsEqualityTests.java

```diff
@@ -0,0 +1,289 @@
+package com.scottlogic.deg.generator.restrictions;
+
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public class SetRestrictionsEqualityTests {
+
+    @Test
+    void equals_whenOtherObjectIsNull_returnsFalse() {
+        SetRestrictions restriction = new SetRestrictions(null, null);
+
+        boolean result = restriction.equals(null);
+
+        Assert.assertFalse(result);
+    }
+
+    @Test
+    void equals_whenOtherObjectIsNotOfTheSameType_returnsTrue() {
+        SetRestrictions restriction = new SetRestrictions(null, null);
+
+        boolean result = restriction.equals("String");
+
+        Assert.assertFalse(result);
+    }
+
+    @Test
+    void equals_whenObjectsAreEqual_returnsTrue() {
+        SetRestrictions restriction1 = new SetRestrictions(null, null);
+        SetRestrictions restriction2 = new SetRestrictions(null, null);
+
+        boolean result = restriction1.equals(restriction2);
+
+        Assert.assertTrue(result);
+    }
+
+    @Test
+    void equals_whenBothSetRestrictionsAreEmptySet_returnsTrue() {
+        SetRestrictions restriction1 = new SetRestrictions(Collections.emptySet(), Collections.emptySet());
+        SetRestrictions restriction2 = new SetRestrictions(Collections.emptySet(), Collections.emptySet());
+
+        boolean result = restriction1.equals(restriction2);
+
+        Assert.assertTrue(result);
+    }
+
+    @Test
+    void equals_whenWhitelistsAreEmptySetAndBlacklistsAreNull_returnsTrue() {
+        SetRestrictions restriction1 = new SetRestrictions(Collections.emptySet(), null);
+        SetRestrictions restriction2 = new SetRestrictions(Collections.emptySet(), null);
+
+        boolean result = restriction1.equals(restriction2);
+
+        Assert.assertTrue(result);
+    }
+
+    @Test
+    void equals_whenBlacklistsAreEmptySetAndWhitelistsAreNull_returnsTrue() {
+        SetRestrictions restriction1 = new SetRestrictions(null, Collections.emptySet());
+        SetRestrictions restriction2 = new SetRestrictions(null, Collections.emptySet());
+
+        boolean result = restriction1.equals(restriction2);
+
+        Assert.assertTrue(result);
+    }
+
+    @Test
+    void equals_whenOneRestrictionWhitelistIsEmptyAndOtherIsNull_returnsFalse() {
+        SetRestrictions restriction1 = new SetRestrictions(Collections.emptySet(), null);
```

Comment:
Does this need to be swapped too, given the name of the test?

Reply:
I think this is what I mean. The whitelist for restriction1 is set to emptySet(), the whitelist for restriction2 is set to null. I have renamed the tests to hopefully better explain.